pandera/checks.py

```diff
@@ -397,9 +397,12 @@ def __call__(
         )
 
     def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            # we can only be equal if the same type
+            return False
```

Comment:
I think we should return the special value [NotImplemented](https://docs.python.org/3.6/library/constants.html#NotImplemented) instead of False.

Reply:
Yep, sounds good.